nativelib/src/main/scala/scala/scalanative/posix/pthread.scala

```diff
@@ -0,0 +1,367 @@
+
+  def pthread_mutexattr_getprioceiling(attr: Ptr[pthread_mutexattr_t],
+                                       prioceiling: Ptr[CInt]): CInt = extern
+
+  def pthread_mutexattr_getprotocol(attr: Ptr[pthread_mutexattr_t],
+                                    protocol: Ptr[CInt]): CInt = extern
+
+  def pthread_mutexattr_getpshared(attr: Ptr[pthread_mutexattr_t],
+                                   pshared: Ptr[CInt]): CInt = extern
+
+  def pthread_mutexattr_getrobust(attr: Ptr[pthread_mutexattr_t],
+                                  robust: Ptr[CInt]): CInt = extern
+
+  def pthread_mutexattr_gettype(attr: Ptr[pthread_mutexattr_t],
+                                tp: Ptr[CInt]): CInt = extern
+
+  def pthread_mutexattr_init(attr: Ptr[pthread_mutexattr_t]): CInt = extern
+
+  def pthread_mutexattr_setprioceiling(attr: Ptr[pthread_mutexattr_t],
+                                       prioceiling: CInt): CInt = extern
+
+  def pthread_mutexattr_setprotocol(attr: Ptr[pthread_mutexattr_t],
+                                    protocol: CInt): CInt = extern
+
+  def pthread_mutexattr_setpshared(attr: Ptr[pthread_mutexattr_t],
+                                   pshared: CInt): CInt = extern
+
+  def pthread_mutexattr_setrobust(attr: Ptr[pthread_mutexattr_t],
+                                  robust: CInt): CInt = extern
+
+  def pthread_mutexattr_settype(attr: Ptr[pthread_mutexattr_t],
+                                tp: CInt): CInt = extern
+
+  def pthread_once(once_control: Ptr[pthread_once_t],
+                   init_routine: routine): CInt = extern
+
+  def pthread_rwlock_destroy(rwlock: Ptr[pthread_rwlock_t]): CInt = extern
+
+  def pthread_rwlock_init(rwlock: Ptr[pthread_rwlock_t],
+                          attr: Ptr[pthread_rwlockattr_t]): CInt = extern
+
+  def pthread_rwlock_rdlock(rwlock: Ptr[pthread_rwlock_t]): CInt = extern
+
+  def pthread_rwlock_timedrdlock(rwlock: Ptr[pthread_rwlock_t],
+                                 timespec: Ptr[timespec]): CInt = extern
+
+  def pthread_rwlock_timedwrlock(rwlock: Ptr[pthread_rwlock_t],
+                                 timespec: Ptr[timespec]): CInt = extern
+
+  def pthread_rwlock_tryrdlock(rwlock: Ptr[pthread_rwlock_t]): CInt = extern
+
+  def pthread_rwlock_trywrlock(rwlock: Ptr[pthread_rwlock_t]): CInt = extern
+
+  def pthread_rwlock_unlock(rwlock: Ptr[pthread_rwlock_t]): CInt = extern
+
+  def pthread_rwlock_wrlock(rwlock: Ptr[pthread_rwlock_t]): CInt = extern
+
+  def pthread_rwlockattr_destroy(attr: Ptr[pthread_rwlockattr_t]): CInt =
+    extern
+
+  def pthread_rwlockattr_getpshared(attr: Ptr[pthread_rwlockattr_t],
+                                    pshared: Ptr[CInt]): CInt = extern
+
+  def pthread_rwlockattr_init(attr: Ptr[pthread_rwlockattr_t]): CInt = extern
+
+  def pthread_rwlockattr_setpshared(attr: Ptr[pthread_rwlockattr_t],
+                                    pshared: CInt): CInt = extern
+
+  def pthread_self(): pthread_t = extern
```

Comment:
It seems like `pthread_t` can be safely treated as `ULong`, so it means that we can pass it by value without a problem.